AEPIdentity/Tests/IdentityTests.swift

```diff
@@ -548,7 +545,25 @@ class IdentityTests: XCTestCase {
         mockRuntime.simulateSharedState(extensionName: "com.adobe.module.analytics", event: getUrlVariablesEvent, data: ([IdentityConstants.Analytics.ANALYTICS_ID: "test-aid"], .set))
 
         // verify
-        XCTAssertFalse(identity.readyForEvent(getUrlVariablesEvent)) // booting up
-        XCTAssertTrue(identity.readyForEvent(getUrlVariablesEvent))
+        XCTAssertTrue(identity.readyForEvent(getUrlVariablesEvent)) // fast boot
+    }
+
+    func testReadyForEventGetEcidOnAppInstallWillWaitForConfigurationAndECIDToBeGenerated() {
+        let getEcidEvent = Event(name: "Test Get ECID Event", type: EventType.identity, source: EventSource.requestIdentity, data: nil)
+
+        mockRuntime.simulateSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: getEcidEvent, data: ([IdentityConstants.Configuration.EXPERIENCE_CLOUD_ORGID: "test-org-id", IdentityConstants.Configuration.GLOBAL_CONFIG_PRIVACY: PrivacyStatus.optedIn.rawValue], .set))
+
+        // verify
+        XCTAssertTrue(identity.readyForEvent(getEcidEvent)) // fast boot
+    }
+
+    func testReadyForEventGetEcidOnWillWaitForConfigurationWhenEcidIsCached() {
```

Comment:
Similar to the above test, can you verify that the event is actually waiting on the configuration state? Maybe something like:

1. set ECID in mock datastore
2. call readyForEvent and verify False
3. verify ECID is set in Identity Properties
4. set valid Configuration
5. call readyForEvent and verify True